### PR TITLE
Update Project Model Test and create new has-attribute helper

### LINKED ISSUE #522: Write unit tests for the Project Model
Write unit tests for the [`Project` Model](https://github.com/code-corps/code-corps-ember/blob/develop/app/models/project.js). There is an [existing unit test file](https://github.com/code-corps/code-corps-ember/blob/develop/tests/unit/models/project-test.js), but it could use more coverage (computed properties & relationship declarations). 

## References
- Progress on: #98 
- [Model testing guide](https://guides.emberjs.com/v2.8.0/testing/testing-models/)
- [Contributing Guidelines](https://github.com/code-corps/code-corps-ember/blob/develop/.github/CONTRIBUTING.md)
- Join our [Slack](https://codecorps.slack.com)

### LINKED ISSUE #554: Write helper for asserting model attributes
# Problem
Model tests for checking attributes are pretty repetitive. To Dry it up we should write a helper for asserting attributes. Ex. `assert.hasAttribute(attributes, 'attribute');`

## References
- [Original Comment](https://github.com/code-corps/code-corps-ember/pull/546#discussion_r81476237)

### PR DESCRIPTION
**Pull request by filipecrosk**
# What's in this PR?

- Update unit test of Project Model
- Create a new assert's helper to validate attributes

## References
Fixes #522 and fixes #554 

Progress on: #98

### CODE REVIEW

File: tests/helpers/has-attributes.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import QUnit from 'qunit';
+
+/**
+  Creates a custom QUnit Assertion.
```
Comment (JoshSmith) [inline tests/helpers/has-attributes.js]: Don't think this line is necessary.

File: tests/helpers/has-attributes.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  This assertion will compare 2 arrays of attributes.
+
+  @method hasAttributes
+  @param {Array} attributes
+  @param {Array} attributesToTest
+*/
+function compareArrays(attributes, attributesToTest) {
+  attributesToTest.sort();
```
Comment (JoshSmith) [inline tests/helpers/has-attributes.js]: I think `attributesToTest` is misleading when we're testing both. I would say the first param should be `actual` and the second `expected`.

File: tests/helpers/has-attributes.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import QUnit from 'qunit';
+
+/**
+  Creates a custom QUnit Assertion.
+  This assertion will compare 2 arrays of attributes.
```
Comment (JoshSmith) [inline tests/helpers/has-attributes.js]: Would mention sorting here.

File: tests/helpers/has-attributes.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+*/
+function compareArrays(attributes, attributesToTest) {
+  attributesToTest.sort();
+
+  return attributes.sort().every(function(element, index) {
+    return element === attributesToTest[index]; 
+  });
+}
+
+QUnit.assert.hasAttributes = function(attributes, attributesToTest) {
```
Comment (JoshSmith) [inline tests/helpers/has-attributes.js]: Would document this.

File: tests/unit/models/project-test.js [hunk 5]
```diff
@@ -1,12 +1,66 @@
...
+    "iconThumbUrl",
+    "longDescriptionBody",
+    "longDescriptionMarkdown",
+    "openTasksCount",
+    "organization",
+    "slug",
+    "title",
+  ];
+
+  assert.hasAttributes(attributes, attributesToTest);
```
Comment (JoshSmith) [inline tests/unit/models/project-test.js]: `actualAttributes, expectedAttributes`

File: tests/unit/models/project-test.js [hunk 6]
```diff
@@ -1,12 +1,66 @@
...
+
+test('it should have open tasks', function(assert) {
+  assert.expect(1);
+
+  let project = this.subject({ openTasksCount: 1 });
+
+  assert.equal(project.get('hasOpenTasks'), true, 'has open tasks');
+});
+
+test('it should have the correct alias from organization-membership', function(assert) {
```
Comment (JoshSmith) [inline tests/unit/models/project-test.js]: Would rename this test something `it should have computed properties for its organization's members`

File: tests/helpers/has-attributes.js [hunk 7]
```diff
@@ -1,24 +1,23 @@
 import QUnit from 'qunit';
-
-/**
-  Creates a custom QUnit Assertion.
+/*
   This assertion will compare 2 arrays of attributes.
+  It first sort both arrays and then compare each element.
```
Comment (JoshSmith) [inline tests/helpers/has-attributes.js]: > sorts, compares

**Comment by filipecrosk** [comment]
@JoshSmith Rebased and Squashed. Can I update the other tests to use this `assert.hasAttributes`?

**Comment by JoshSmith** [comment]
@filipecrosk go for it!

**Pull request merged**